test/integration/targets/inventory_vmware_host_inventory/test_vmware_host_inventory.yml

```diff
@@ -0,0 +1,20 @@
+# Test code for the vmware host dynamic plugin module
+# Copyright: (c) 2019, Abhijeet Kasurde <akasurde@redhat.com>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+---
+- name: Test VMware Host Dynamic Inventroy Plugin
+  hosts: localhost
+  tasks:
+    - name: store the vcenter container ip
+      set_fact:
+        vcsim: "{{ lookup('env', 'vcenter_host') }}"
+
+    - debug:
+        var: vcsim
+
+    - name: Check that there are 'all' groups present in inventory
+      assert:
+        that: "'{{ item }}' in {{ groups.keys() | list }}"
```

Comment:
This fails on python 2.7, but this code is never ran because of the ls problem mentioned in the other comment. 
See [my changes](https://github.com/ansible/ansible/pull/52782/files#diff-d6070f65d91ac9d1b70aacdeea5828a7R15) on an identical code on PR #52782 .